pkg/plugins/common/kustomize/v1/init.go

```diff
@@ -94,6 +108,35 @@ func (p *initSubcommand) InjectConfig(c config.Config) error {
 	return nil
 }
 
+func (p *initSubcommand) PreScaffold(machinery.Filesystem) error {
+	arch := runtime.GOARCH
+	// It probably will never return x86_64. However, we are here checking the support for the binaries
+	// So that, x86_64 means getting the Linux/amd64 binary. Then, we just keep this line to ensure
+	// that it complies with the same code implementation that we have in the targets. In case someone
+	// call the command inform the GOARCH=x86_64 then, we will properly handle the scenario
+	// since it will work successfully and will instal the Linux/amd64 binary via the Makefile target.
+	arch = strings.Replace(arch, "x86_64", "amd64", -1)
+	localPlatform := fmt.Sprintf("%s/%s", strings.TrimSpace(runtime.GOOS), strings.TrimSpace(arch))
+	found := false
+	for _, value := range supportedArchs {
+		if value == localPlatform {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		log.Warnf("the platform of this environment (%s) is not suppported by kustomize v3 (%s) which is "+
+			"used in this scaffold. You will be unable to download a binary for the kustomize version supported "+
+			"and used by this plugin. The currently supported platforms are: %q",
+			localPlatform,
+			KustomizeVersion,
+			supportedArchs)
+	}
+
+	return nil
```

Comment:
It seems the error return of this function will always be nil?

Reply:
Yes, because we will never return an error here so far.
See that it is an implementation of the interface:

```
// PreScaffold executes tasks before the main scaffolding.
	PreScaffold(machinery.Filesystem) error
```